dashboard/app/models/course_offering.rb

```diff
@@ -240,6 +240,21 @@ def can_be_assigned?(user)
     false
   end
 
+  # Return if the course offering's device compatibilities JSON string either:
+  # - Is null
+  # - Contains a device compatibility mapping to the (None) option (which would appear like `{tablet: ""}`)
+  def missing_device_compatibility?
+    return true unless device_compatibility
+
+    device_compatibility_values = JSON.parse(device_compatibility).values
+    device_compatibility_values.any?(&:blank?)
+  end
+
+  def missing_required_device_compatibility?
+    # Course offering device compatibilites are required to publish student courses that are assignable.
+    assignable? && get_participant_audience == 'student' && missing_device_compatibility?
```

Comment:
I wonder if we shouldn't even check `assignable`? Sometimes, the course offering is made assignable after the course/unit is published and I don't think we have protections there.

Reply:
Sounds good! On it!